test/cycle.js

```diff
@@ -134,4 +139,115 @@ describe('Cycle', function () {
       }, 10);
     });
   });
+
+  describe('isolate()', function () {
+    it('should throw if first argument is not a function', function () {
+      assert.throws(() => {
+        Cycle.isolate('not a function');
+      }, /First argument given to Cycle\.isolate\(\) must be a 'dialogue' function/i);
+    });
+
+    it('should throw if second argument is not a string', function () {
+      function MyDialogue() {}
+      assert.throws(() => {
+        Cycle.isolate(MyDialogue, null);
+      }, /Second argument given to Cycle\.isolate\(\) must be a string for 'scope'/i);
+    });
+
+    it('should return a function', function () {
+      function MyDialogue() {}
+      const scopedMyDialogue = Cycle.isolate(MyDialogue, `myScope`);
+      assert.strictEqual(typeof scopedMyDialogue, `function`);
+    });
+
+    it('should make a new scope if second argument is undefined', function () {
+      function MyDialogue() {}
+      const scopedMyDialogue = Cycle.isolate(MyDialogue);
+      assert.strictEqual(typeof scopedMyDialogue, `function`);
+    });
+
+    describe('scopedDialogue', function () {
+      it('should return scoped sinks', function () {
```

Comment:
This is actually not testing scoping, so I'd just rename this as `it('should return a valid dialogue')`